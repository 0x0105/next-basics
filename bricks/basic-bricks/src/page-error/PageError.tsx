import React from "react";
import { useTranslation } from "react-i18next";
import { NS_BASIC_BRICKS, K } from "../i18n/constants";

interface PageErrorProps {
  error: string;
}

export function PageError(props: PageErrorProps): React.ReactElement {
  const { t } = useTranslation(NS_BASIC_BRICKS);

  return (
    <div style={{ textAlign: "center", marginTop: 100 }}>
      <p style={{ fontSize: "150%" }}>{t(K.PAGE_ERROR_TITLE)}</p>
      <p>{props.error}</p>
    </div>
  );
}
