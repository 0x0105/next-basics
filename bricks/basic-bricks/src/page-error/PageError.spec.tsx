import React from "react";
import { shallow } from "enzyme";
import { PageError } from "./PageError";

describe("PageError", () => {
  it("should work", () => {
    const wrapper = shallow(<PageError error="oops" />);
    expect(wrapper).toMatchSnapshot();
  });
});
