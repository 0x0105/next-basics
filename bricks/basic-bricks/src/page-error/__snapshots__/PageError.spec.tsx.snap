// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageError should work 1`] = `
<div
  style={
    Object {
      "marginTop": 100,
      "textAlign": "center",
    }
  }
>
  <p
    style={
      Object {
        "fontSize": "150%",
      }
    }
  >
    PAGE_ERROR_TITLE
  </p>
  <p>
    oops
  </p>
</div>
`;
